src/components/AssignmentTime/ExcecutionTimeMenu.tsx

```diff
@@ -0,0 +1,326 @@
+  const from = fromDate ? fortmatIsoStringToLocalTime(fromDate) : null
+  const to = toDate ? fortmatIsoStringToLocalTime(toDate) : null
+  return (
+    <span>
+      {from} {to ? '-' : ''} {to}
+    </span>
+  )
+}
+
+const testValid = (time: string): boolean => {
+  return (/^([0-1]?[0-9]|2[0-4]):([0-5][0-9])(:[0-5][0-9])?$/.test(time))
+}
+
+export const ExcecutionTimeMenu = ({ handleOnSelect, index, startDate }: ExcecutionTimeItemsProps): JSX.Element => {
+  const [open, setOpen] = useState(false)
+  const inputRef = useRef(null)
+  const [data] = useYValue<AssignmentData>(`meta.core/assignment[${index}].data`)
+  const [selected, setSelected] = useState<Date[]>([new Date(`${startDate}T00:00:00`)])
+  const [startTimeValue, setStartTimeValue] = useState<string>('00:00')
+  const [endTimeValue, setEndTimeValue] = useState<string>('23:59')
+  const [startDateValue, setStartDateValue] = useState<string>()
+  const [endDateValue, setEndDateValue] = useState<string>()
+  const [hasEndTime, setHasEndTime] = useState<boolean>()
+  const { locale, timeZone } = useRegistry()
+  const [mounted, setMounted] = useState(false)
+  const [startTimeValid, setStartTimeValid] = useState(false)
+  const [endTimeValid, setEndTimeValid] = useState(false)
+
+  useEffect(() => {
+    if (!mounted && data) {
+      const dates: Date[] = []
+
+      if (data?.start) {
+        const aDate = new Date(data.start.toString())
+        const startValue = aDate.toLocaleString('sv-SE', {
+          hour: '2-digit',
+          minute: '2-digit'
+        })
+        setStartTimeValue(startValue)
+        setStartDateValue(data.start)
+        setStartTimeValid(testValid(startValue))
+
+        const startDateObject = new Date(data.start)
+        const newStartDayWithTime = new Date(
+          startDateObject.getFullYear(),
+          startDateObject.getMonth(),
+          startDateObject.getDate(),
+          0,
+          0,
+          0,
+          0
+        )
+        dates.push(newStartDayWithTime)
+      }
+
+      if (data?.end) {
+        const aDate = new Date(data.end.toString())
+        const endValue = aDate.toLocaleString('sv-SE', {
+          hour: '2-digit',
+          minute: '2-digit'
+        })
+        setEndTimeValue(endValue)
+        setEndDateValue(data.end)
+        setHasEndTime(true)
+        setEndTimeValid(testValid(endValue))
+
+        const endDate = new Date(data.end)
+        const newEndDayWithTime = new Date(
+          endDate.getFullYear(),
+          endDate.getMonth(),
+          endDate.getDate(),
+          0,
+          0,
+          0,
+          0
+        )
+        if (dates.length === 1 && (dates[0].getTime() !== newEndDayWithTime.getTime())) {
+          dates.push(newEndDayWithTime)
+        }
+      }
+
+      setSelected(dates)
+      setMounted(true)
+    }
+  }, [data, mounted])
+
+  const handleStartTimeChange: ChangeEventHandler<HTMLInputElement> = (e) => {
```

Comment:
Feels like there is some common grounds here to reduce the code an increase readability
Maybe something along the lines of: (Not sure if its working)

```
const handleTimeChange = (
  time: string,
  selectedDate: Date,
  setDateValue: (value: string) => void
): void => {
  const [hours, minutes] = time.split(':').map((str: string) => parseInt(str, 10))
  const newSelectedDate = new Date(selectedDate)
  newSelectedDate.setHours(hours, minutes)
  setDateValue(newSelectedDate.toISOString())
}

const handleStartTimeChange: ChangeEventHandler<HTMLInputElement> = (e) => {
    const time = e.target.value
    handleTimeChange(time, selected[0], setStartDateValue)
    setStartTimeValue(time)
  }

  const handleEndTimeChange: ChangeEventHandler<HTMLInputElement> = (e) => {
    const time = e.target.value
    handleTimeChange(time, selected.length === 2 ? selected[1] : selected[0], setEndDateValue)
    setEndTimeValue(time)
  }

```